### DISCUSSION TITLE
寻求 troubleshooting 系统 hang 问题帮助

### DISCUSSION THREAD

**Discussion by guyuming76** [fcitx/fcitx5#1123] [Q&A]
我最近在使用fcitx5 和基于wlroots的wayland合成器 dwl 的时候经常遇到系统 hang 的情况,可以频繁到每天重启几次.

为了找原因, 我先启动xorg, 然后使用x11 backend 启动dwl, 这样, dwl hang了的时候, 我可以在xorg里打开一个终端模拟器. 我发现, 只要我 kill 掉fcitx5进程,dwl就会恢复响应.

gdb attach 到dwl进程,查看bt,发现栈顶总是printstatus 函数里的printf 输出状态的语句. 我若是把printstatus函数内容注释掉重新编译安装dwl,就没有遇到过hang的情况.

但试图 gdb attach 到fcitx5 进程总是不能成功.

htop里看dwl进程的文件0和fcitx5进程的文件1都对应/dev/pts/0; 两者的文件1和0又都是pipe,mode数字相同

lslocks 命令我也没看到啥死锁信息.

我该如何继续查找原因呢?

**Comment by wengxt** [comment]
gdb attach 到 fcitx 总是不能成功是什么意思

debug fcitx 需要注意的是正在使用的终端的按键是否也被发送给 fcitx
 
https://fcitx-im.org/wiki/Debug_fcitx5 列出了一些阻止终端使用输入法的方式

**Reply by guyuming76** [reply]
哦,把这个给忘了:(
不过这个方法好麻烦的,我只是要看下hang住时候的调用栈, 下次遇到这个情况的时候试下 pstack 脚本看行不行
